clis/daemon-common/src/daemon.rs

```diff
@@ -19,21 +19,23 @@ pub enum ProcessType {
 /// Struct for packaging together the functions for managing the daemon.
 /// Currently does not need any members, since most of the data is stored
 /// in files anyway.
-pub struct Daemon;
+pub trait Daemon {
+    /// Daemon name, used for naming daemon files
+    const NAME: &'static str;
 
-impl Daemon {
     /// Checks whether the daemon is currently running.
     /// Won't work if pid file had been tampered with or it's location changed.
     ///
     /// # Returns
     ///
     /// Result containing a bool. True means that the daemon is running.
-    pub fn is_running() -> Result<bool> {
+    fn is_running() -> Result<bool> {
```

Comment:
I have not yet double-checked this, but I wonder, maybe it is possible to check `.is_running` by the existance or non-existance of unix socket?

Note that in OpenWRT, while we can have `teliod start` and `teliod stop` commands, we also need ability to start teliod process in  daemon mode directly, and then the client should fail if unix socket does not exist instead of spawning new daemon. This is because, `procd` is responsible for process management in OpenWRT, and I guess it would be nicer to cooperate with it, instead of spawning "unmanaged" processes.

Reply:
Hmm, is checking it by the pid worse than checking it by the socket? Sure, I will add some option to spawn it directly.

Reply:
I was thinking about, because generally we would have two kind of indicators about existance of daemon (pid file and unix socket) therefore I thought this may be a bit redundant, thats all. But fundamentally - no problems, maybe just that we should store the pid file somewhere in standartized location, e.g. like `/var/run` or `/run` or similar.